packer/plugin-getter/plugins.go

```diff
@@ -826,11 +813,66 @@ func (pr *Requirement) InstallLatest(opts InstallOptions) (*Installation, error)
 							break
 						}
 						if copyFrom == nil {
-							err := fmt.Errorf("could not find a %s file in zipfile", checksum.Filename)
+							err := fmt.Errorf("could not find a %q file in zipfile", expectedBinaryFilename)
 							errs = multierror.Append(errs, err)
 							return nil, errs
 						}
 
+						tempPluginPath := filepath.Join(os.TempDir(), fmt.Sprintf(
+							"packer-plugin-temp-%s%s",
+							random.Numbers(8),
+							opts.BinaryInstallationOptions.Ext))
+
+						// Save binary to temp so we can ensure it is really the version advertised
+						tempOutput, err := os.OpenFile(tempPluginPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+						if err != nil {
+							log.Printf("[ERROR] failed to create temp plugin executable: %s", err)
+							return nil, multierror.Append(errs, err)
+						}
+						defer os.Remove(tempPluginPath)
+
+						_, err = io.Copy(tempOutput, copyFrom)
+						if err != nil {
+							log.Printf("[ERROR] failed to copy uncompressed binary to %q: %s", tempPluginPath, err)
+							return nil, multierror.Append(errs, err)
+						}
+
+						// Not a problem on most platforms, but unsure Windows will let us execute an already
+						// open file, so we close it temporarily to avoid problems
+						_ = tempOutput.Close()
+
+						desc, err := GetPluginDescription(tempPluginPath)
+						if err != nil {
+							err := fmt.Errorf("failed to describe plugin binary %q: %s", tempPluginPath, err)
+							errs = multierror.Append(errs, err)
+							continue
+						}
+
+						descVersion, err := goversion.NewSemver(desc.Version)
+						if err != nil {
+							err := fmt.Errorf("invalid self-reported version %q: %s", desc.Version, err)
+							errs = multierror.Append(errs, err)
+							continue
+						}
+						if descVersion.Prerelease() != "" {
```

Comment:
This code is also adding hardening to the installation of prereleases, right? Should this check be after the check of versions?

Reply:
Pre-release remote installation is not supported at the moment, only local installation is through `packer plugins install --path`.
Though with Wilken's last remark, I guess this may change so that the version remotely installed could be a `dev` one, albeit with a warning that there's a mismatch.
But in any case, describing a constraint with `-dev` in the version, or installing pre-releases through `packer plugins install` without passing in a path is not a supported workflow for the moment.